lib/mongo/server/connection_pool.rb

```diff
@@ -148,6 +148,16 @@ def initialize(server, options = {})
         @populator = Populator.new(self, options)
         @populate_semaphore = Semaphore.new
 
+        # Condition variable to enforce the first check in check_out: max_pool_size.
+        # This condition variable should be signaled when the number of
+        # unavailable connections decreases (pending + pending_connections +
+        # available_connections).
+        @size_cv = Mongo::ConditionVariable.new(@lock)
+        # This represents the number of threads that have made it past the size_cv
+        # gate but have not acquired a connection to add to the pending_connections
+        # set.
+        @pending = 0
```

Comment:
```suggestion
        @pending_threads_count = 0
```

Reply:
I changed this to `@connection_requests` in the next PR if that's okay

Reply:
👍